src/main/java/org/calrissian/flowmix/api/aggregator/FunctionAggregator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2015 Calrissian.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.calrissian.flowmix.api.aggregator;
+
+/**
+ * All purpose function aggregator
+ *
+ * @author Miguel A. Fuentes Buchholtz
+ * @param <T> result type
+ * @param <F> Field type
+ */
+public class FunctionAggregator<T, F> extends AbstractAggregator<T, F> {
```

Comment:
Agree, this is not done and maybe doomed I was thinking of a more fast way to define operations but the abstract agg does the jog, I'm going to see how to not pull request this maybe-bad-codes